content/blog/en/building-dojo-app.md

```diff
@@ -0,0 +1,1124 @@
+---
+title: Building apps with Dojo
+date: 2020-11-30T12:00:00.000Z
+author: Rene Rubalcava
+---
+
+Dojo provides a one stop shop for building scalable and efficient web applications. It comes with all the basics you need:
+
+* Widgets
+* Routing
+* State Management
+
+Sometimes when building an app, you might have to mix and match different libraries to get all these features, but when working with a full framework, this is the type of tooling you would expect to get out of the box. It's a tradeoff of flexibility versus versatility. I personally like having everything I need in one place, with room to bring in extra tooling as needed.
+
+Today, we're going to build a coffee shop app, where you can add a type of coffee and customize it for your order. These are some of the key items and features our application will need.
+
+* Home Page
+* Menu Page
+* Drink Page to customize order
+* Add to Cart
+* Remove from Cart
+
+Navigating between these pages will require the use of routing and we'll use Dojo state management to handle customizing drink orders.
+
+## Start with state
+
+I like to start writing my apps by defining interfaces and types. This is why I'm such a huge fan of TypeScript, it let's me take a look at an application at a high level and think about it from a data perspective. Since this is a coffee shop application, we should probably have an interface for our coffee drink.
```

Comment:
```suggestion
Let's start by defining interfaces and types. This is why I'm such a huge fan of TypeScript, it let's me take a look at an application at a high level and think about it from a data perspective. Since this is a coffee shop application, we should probably have an interface for our coffee drink.
```